### PR TITLE
DOC: show failing string on numeric parse

### PR DESCRIPTION
**Pull request by chris-b1**
- [ ] related to #13746
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry not needed?

``` python
# before
In [7]: pd.to_numeric(['a'])
ValueError: Unable to parse string

# after
In [1]: pd.to_numeric(['a'])
ValueError: Unable to parse string "a" at position 0
```

### CODE REVIEW

**Comment by jreback** [comment]
I think its straightforward to also provide a row/line number here, e.g. ``i``

File: pandas/src/inference.pyx [hunk 1]
```diff
@@ -686,9 +686,9 @@ def maybe_convert_numeric(object[:] values, set na_values,
                             raise ValueError('integer out of range')
                     else:
                         seen_float = True
-            except:
+            except (TypeError, ValueError) as e:
                 if not coerce_numeric:
-                    raise
+                    raise type(e)(e.message + ' in row {}'.format(i))
```
Comment (jreback) [inline pandas/src/inference.pyx]: acutally maybe row is not the word, these are 1-d arrays, so maybe entry? value at index i?

Comment (jorisvandenbossche) [inline pandas/src/inference.pyx]: Sorry to do a bit purist, but 'array index' sounds also a bit strange when you pass a Series and not an array. And given that just 'at index' can be confused with the actual index labels, maybe just 'at position ..' ?

Comment (chris-b1) [inline pandas/src/inference.pyx]: Yeah, I do think that's better.

**Comment by chris-b1** [comment]
@jorisvandenbossche -  I think this one is good now.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**